### ISSUE TITLE
Does not work well in mobile devices

### ISSUE THREAD

**Issue by AviKKi** [AviKKi/hover-preview#10]
mouseenter event is only registered on a tap, using [pointerenter](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/pointerenter_event) and [pointerleave](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/pointerleave_event) may help

**Comment by yhau1989** [comment]
pointerenter and pointerleave [is not supported ](https://caniuse.com/pointer)in Safari/Safari IOS

**Comment by AviKKi** [comment]
@yhau1989 can you suggest a solution? If you want to solve this I can assign it to you.

**Comment by yhau1989** [comment]
in mobile the UX concept is touch (click) or slider automatic

**Comment by AviKKi** [comment]
I get what you are saying, it's how mobile YouTube app shows preview on scrolling, the video on top  automatically starts preview.

But it won't work for my use-case, we are using 2 column grid. A expected behavior on mobile is when user touches(not click) over a thumbnail, it should start the preview.

A touch and a click are different, if you tap on the screen it's a click but if you touch and hold for a bit longer, it won't trigger click event; we want to trigger preview in this case.

As library grows we can add more configurations, but for now I'm focusing on this.

PS: My client has suggested that pornhub.com has this type of UX for thumbnail preview

**Comment by yhau1989** [comment]
@AviKKi haha pornhub.com is good case study, this events is activated when scroll is moving up or down, used id # by example <div id="#picture1">

**Comment by AviKKi** [comment]
I'll dig into what event listeners they are using when I get the time. Your suggestion makes a lot sense.

**Comment by yhau1989** [comment]
I am work in a demo 🕙

**Comment by yhau1989** [comment]
my proposal is done , [check out here](https://yhau1989.github.io/hover-preview/example/showcase.html) in your📱

**Comment by AviKKi** [comment]
@yhau1989  [this](https://yhau1989.github.io/hover-preview/example/showcase.html) doesn't seem to be working

Edit -
I had to resize my window for it to work.

I'm summing up your approach for future reference.

- store y coordinate of all the img elements
- when scrolled check the img that coincides with +-10 range for `window.scrollY`, if found start it's preview.

my remarks and opinion -
- When there are multiple column, only the image on last column will work, as only one preview works at a time, last will override previous ones.
- Images at the bottom of a page won't work, this will work best for infinite scroll behavior.
- Instead of looping through images, we can optimize it with binary search / set (but there is payload overhead).

PS: This seems more like youtube app's logic, I'll reverse pornhub's code later(when I get some :alarm_clock: ) to see how they are doing it.

**Comment by yhau1989** [comment]
Of course, my change is focus in mobile version, but responsive design needs to be improved
